src/renderers/webgl/WebGLAttributes.js

```diff
@@ -121,10 +122,20 @@ function WebGLAttributes( gl ) {
 
 	}
 
-	function update( attribute, bufferType ) {
+	function update( geometry, attribute, bufferType ) {
 
 		if ( attribute.isInterleavedBufferAttribute ) attribute = attribute.data;
 
+		var refs = attributeReferences.get( geometry );
+
+		if ( ! refs ) {
+
+			attributeReferences.set( geometry, refs = new Set() );
```

Comment:
nit: I haven't seen this code style elsewhere.

    refs = new Set();
    attributeReferences.set( geometry, refs );

would be better?

Reply:
Can we really use `Set`? It's no part of the core so far...

Reply:
I haven't read the entire code in this PR yet but I think it's better to avoid using it if possible. If we need to use, we should check the performance first.